src/service/performance-impl.js

```diff
@@ -39,15 +42,43 @@ let TickEventDef;
 /**
  * @enum {number}
  */
-export const LCP_ELEMENT_TYPE = {
+export const ELEMENT_TYPE = {
   other: 0,
-  image: 1,
-  video: 2,
-  ad: 3,
-  carousel: 4,
-  bcarousel: 5,
+  image: 1 << 0,
+  video: 1 << 1,
+  ad: 1 << 2,
+  carousel: 1 << 3,
+  bcarousel: 1 << 4,
+  text: 1 << 5,
 };
 
+/**
+ * @param {!Node} node
+ * @return {ELEMENT_TYPE}
+ */
+function getElementType(node) {
+  const {tagName} = getOutermostAmpElement(node);
+  if (tagName == null) {
```

Comment:
WDYT about a separate object/map from tagName to `ELEMENT_TYPE` so this would work out to be more like `return tagTypeMap[getOutermostAmpElement(node)] ?? ELEMENT_TYPE.other` rather than all these ifs? a la
```javascript
const tagTypeMap = {
  [null]: ELEMENT_TYPE.text,
  'IMG': ELEMENT_TYPE.image,
  'AMP-IMG': ELEMENT_TYPE.image,
  'VIDEO': ELEMENT_TYPE.video,
  'AMP-VIDEO': ELEMENT_TYPE.video,
  'AMP-CAROUSEL': ELEMENT_TYPE.carousel,
  'AMP-BASE-CAROUSEL': ELEMENT_TYPE.bcarousel,
  'AMP-AD': ELEMENT_TYPE.ad,
}
```

Reply:
Object lookup is slower than if statements.